tests/array/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -8,3 +8,41 @@ numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9]
 numbers
 
 `;
+
+exports[`string_array.rb 1`] = `
+def strings
+  %w(
+    a
+    b
+    c
+    d
+  )
+end
+
+strings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+def strings
+  ['    a', '    b', '    c', '    d']
+end
+strings
+
+`;
+
+exports[`symbol_array.rb 1`] = `
+def symbols
+  %i(
+    a
+    b
+    c
+    d
+  )
+end
+
+symbols
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+def symbols
+  ['    a', '    b', '    c', '    d']
```

Comment:
It seems that the Ripper sexpressions from Ruby include this whitespace:

```ruby
[:program,
 [[:def,
   [:@ident, "strings", [1, 4]],
   [:params, nil, nil, nil, nil, nil, nil, nil],
   [:bodystmt,
    [[:array,
      [[:@tstring_content, "    a", [3, 0]],
       [:@tstring_content, "    b", [4, 0]],
       [:@tstring_content, "    c", [5, 0]],
       [:@tstring_content, "    d", [6, 0]]]]],
    nil,
    nil,
    nil]],
  [:vcall, [:@ident, "strings", [10, 0]]]]]
```

But if you run it in an interpreter the white space isn't present:

```
>   strings
=> ["a", "b", "c", "d"]
```

I'm not sure what to make of this, other than it's not quite the behavior I expected 🤔

Reply:
The sexp is correct, if you check the tokens array, you will find one of these symbols.

```ruby
:on_qwords_beg, :on_qsymbols_beg, :on_words_beg, :on_symbols_beg
```

What needs to be done is to keep on removing tokens from the tokens array wherever the need arises. So inside visit_array func we can check what is the current token and based on that we can format the `:@tstring_content` value to strip the whitespace if the current_token is `:on_words_beg` and so on

Reply:
Creating a separate issue to keep track of this behavior:

https://github.com/iamsolankiamit/prettier-ruby/issues/6